### PR TITLE
feat(design-system): Add data-feature to Modal

### PR DESCRIPTION
**Pull request by tlnd-mhuchet**
**What is the problem this PR is trying to solve?**
The `Modal` component define `data-test` and `data-testid` attributes, but no `data-feature`.

The `data-feature` attribute is used by PMs in Pendo to track events related to modals.

**What is the chosen solution to this problem?**
Add a new props `dataFeature`. Use it to identify the whole modal, and as a prefix to identify all actions.

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/design-system/src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -33,12 +33,13 @@ export type ModalPropsType = {
...
 	};
-	onClose?: Function;
+	onClose?: () => void;
 	disclosure?: ReactElement;
 	primaryAction?: PrimaryActionPropsType;
 	secondaryAction?: ButtonSecondaryPropsType;
 	preventEscaping?: boolean;
 	children: ReactNode | ReactNode[];
+	dataFeature?: string;
 };
```
Comment (Benrajalu) [inline packages/design-system/src/components/Modal/Modal.tsx:43]:
```suggestion
} & Omit<HTMLAttributes<HTMLDivElement>, 'className' | 'style'>;
```
We should be able to spread whatever data-thing we want on there :)

File: packages/design-system/src/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -52,8 +53,16 @@ function PrimaryAction(props: PrimaryActionPropsType) {
...
-		props;
+	const {
+		header,
+		primaryAction,
+		disclosure,
+		onClose,
+		secondaryAction,
+		preventEscaping,
+		children,
+		dataFeature = 'default-modal',
```
Comment (Benrajalu) [inline packages/design-system/src/components/Modal/Modal.tsx:64]:
```suggestion
		...rest,
```

File: packages/design-system/src/components/Modal/Modal.tsx [hunk 3]
```diff
@@ -80,6 +89,7 @@ function Modal(props: ModalPropsType): ReactElement {
 						<Dialog
 							{...dialog}
 							data-test="modal"
+							data-feature={dataFeature}
```
Comment (Benrajalu) [inline packages/design-system/src/components/Modal/Modal.tsx:92]:
```suggestion
							{...dialog}
							{...rest}
							data-test="modal"
```

**Review by Benrajalu — APPROVED** [review]

File: packages/design-system/src/components/Modal/Modal.tsx [hunk 4]
```diff
@@ -123,11 +133,19 @@ function Modal(props: ModalPropsType): ReactElement {
 										</span>
 
 										{secondaryAction && (
-											<ButtonSecondary {...secondaryAction} data-testid="modal.buttons.secondary" />
+											<ButtonSecondary
+												{...secondaryAction}
+												data-testid="modal.buttons.secondary"
+												data-feature="modal.buttons.secondary"
```
Comment (romainseb) [inline packages/design-system/src/components/Modal/Modal.tsx:139]: Can't we pass the data-feature we want to with the spread of secondaryAction driven by the host ?

File: packages/design-system/src/components/Modal/Modal.tsx [hunk 5]
```diff
@@ -123,11 +133,19 @@ function Modal(props: ModalPropsType): ReactElement {
...
+											/>
 										)}
 
 										{primaryAction && (
-											<PrimaryAction {...primaryAction} data-testid="modal.buttons.primary" />
+											<PrimaryAction
+												{...primaryAction}
+												data-testid="modal.buttons.primary"
+												data-feature="modal.buttons.primary"
```
Comment (romainseb) [inline packages/design-system/src/components/Modal/Modal.tsx:147]: Can't we pass the data-feature we want to with the spread of primaryAction driven by the host ?

Comment (jmfrancois) [reply packages/design-system/src/components/Modal/Modal.tsx:147]: so why not add it to types ? it make sens, we are adding data attribute on user actions

Comment (tlnd-mhuchet) [reply packages/design-system/src/components/Modal/Modal.tsx:146]: Created local type for data attributes and added it to ButtonPrimitive props.

File: packages/design-system/src/types/index.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export type DataAttributes = {
+	'data-feature'?: string | undefined;
+	'data-test'?: string | undefined;
+	'data-testid'?: string | undefined;
```
Comment (Benrajalu) [inline packages/design-system/src/types/index.ts:4]:
```suggestion
	'data-feature'?: string;
	'data-test'?: string;
	'data-testid'?: string;
```
If one choses to use a data-attribute, they should not set it to undefined

**Pull request merged**